### PR TITLE
Added printing in contracts

### LINKED ISSUE #481: Enable printing in contracts
Make contracts able to print

### PR DESCRIPTION
**Pull request by Radinyn**
Closes #481

## Introduced changes

To use this functionality this has to be included in `Scarb.toml`:

```toml
[[target.starknet-contract]]
allowed-libfuncs-deny = false
```

Effect: 
image

## Checklist

- [x] Linked relevant issue
- [ ] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [ ] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/cheatnet/src/contract_print.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        let inputs = match extract_input(vm, input_start, input_end) {
+            Ok(inputs) => inputs,
+            Err(err) => return PrintingResult::Err(err),
+        };
+
+        match selector {
+            "print" => {
+                for value in inputs {
+                    if let Some(short_string) = as_cairo_short_string(&value) {
+                        println!("PRINT: {value} => {short_string}",);
```
Comment (war-in) [inline crates/cheatnet/src/contract_print.rs:51]:
```suggestion
                        println!("PRINT: {value} => {short_string}");
```

**Review by war-in — APPROVED** [review]
Please include the docs 🙏

File: crates/cheatnet/src/contract_print.rs (hunk 1 above)
Comment (Radinyn) [reply crates/cheatnet/src/contract_print.rs]: I will include the docs in a follow-up PR

File: crates/forge/tests/data/contract_printing/Scarb.toml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[package]
+name = "contract_printing"
+version = "0.1.0"
+
+# See more keys and their definitions at https://docs.swmansion.com/scarb/docs/reference/manifest
+
+[[target.starknet-contract]]
+sierra = true
+casm = true
+allowed-libfuncs-deny = false
```
Comment (piotmag769) [inline crates/forge/tests/data/contract_printing/Scarb.toml:10]: It does not compile without it?

**Pull request merged**